cli/pcluster/config/validators.py

```diff
@@ -1253,3 +1262,30 @@ def fsx_ignored_parameters_validator(section_key, section_label, pcluster_config
 def _describe_ec2_key_pair(key_pair_name):
     """Return information about the provided ec2 key pair."""
     return boto3.client("ec2").describe_key_pairs(KeyNames=[key_pair_name])
+
+
+def ebs_volume_type_size_validator(section_key, section_label, pcluster_config):
+    """
+    Validate that the EBS volume size matches the chosen volume type.
+
+    The default value of volume_size for EBS volumes is 20 GiB.
+    The volume size of standard ranges from 1 GiB - 1 TiB
+    The volume size of gp2 ranges from 1 GiB - 16 TiB
+    The volume size of io1 and io2 ranges from 4 GiB - 16 TiB
+    The volume sizes of st1 and sc1 range from 500 GiB - 16 TiB
+    """
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section(section_key, section_label)
+    volume_size = section.get_param_value("volume_size")
+    volume_type = section.get_param_value("volume_type")
+
+    if volume_type in EBS_VOLUME_TYPE_TO_VOLUME_SIZE_BOUNDS:
+        min_size, max_size = EBS_VOLUME_TYPE_TO_VOLUME_SIZE_BOUNDS.get(volume_type)
+        if volume_size > max_size:
+            errors.append("The volume size of {0} must be at most {1} TiB".format(volume_type, int(max_size / 1024)))
```

Comment:
let's use units of GiB in the error message since that's what is used in the config file. that way users don't have to do the conversion themselves.

Reply:
Make sense! It is done and I also make change to the comments of this function